marvin.py

```diff
@@ -603,14 +647,34 @@ def main(self):
         except FileNotFoundError:
             self.logger.error("FATAL ERROR-->" + self.comment_file_name + " FILE NOT FOUND, ABORTING...")
             quit(1)
+
         # Read the rules used to delete a post
         try:
             with open(self.rules_file_name) as data_file:
                 rules_list = json.load(data_file)
                 for current_rule in rules_list["rules"]:
                     self.rules[current_rule["number"]] = current_rule["text"]
         except FileNotFoundError:
-            self.logger.error("FATAL ERROR-->" + self.config_file_name + " FILE NOT FOUND, ABORTING...")
+            self.logger.error("FATAL ERROR-->" + self.rules_file_name + " FILE NOT FOUND, ABORTING...")
+            quit(1)
+
+        # Read the blacklisted words
+        try:
+            with open(self.word_blacklist_file_name) as data_file:
+                word_blacklist2 = json.load(data_file)
+                for current_word in word_blacklist2["words"]:
+                    self.word_blacklist.append(current_word)
+        except FileNotFoundError:
+            self.logger.error("FATAL ERROR-->" + self.word_blacklist_file_name + " FILE NOT FOUND, ABORTING...")
+            quit(1)
+        self.word_blacklist.sort()
```

Comment:
Perché le ordini?

Reply:
In realtà scritta così ha poco senso. Mi stava sulle palle che la funzione di controllo delle parole in blacklist avesse complessità n^2. Così ho pensato: se riordino entrambi gli insiemi (in teoria complessità n log n) posso fare un confronto più facile di complessità n. Devo fare il sort anche su words e togliere il doppio for nel controllo, lo committo e così ci sarà maggior risparmio prestazionale.

Reply:
fatto